dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/EnrollmentDeletionHandler.java

```diff
@@ -60,15 +58,10 @@ private DeletionVeto allowDeleteProgram(Program program) {
   }
 
   private void deleteProgram(Program program) {
-    Collection<Enrollment> enrollments = enrollmentService.getEnrollments(program);
-
-    if (enrollments != null) {
-      Iterator<Enrollment> iterator = enrollments.iterator();
-      while (iterator.hasNext()) {
-        Enrollment enrollment = iterator.next();
-        iterator.remove();
-        enrollmentService.hardDeleteEnrollment(enrollment);
-      }
+    if (program.isRegistration()) {
+      return;
     }
+
+    delete("delete from enrollment where programid = :id", Map.of("id", program.getId()));
```

Comment:
this is deleting dummy enrollments right?
is it correct that other enrollments into a program with registration are only deleted by trackers TrackerObjectDeletionService?

I probably got all of the above wrong which suggests we should document this at least in a comment here.

Reply:
you've got it right, but I can add a comment anyway

Reply:
😅 yes please